stl/inc/format

```diff
@@ -4354,6 +4356,268 @@ public:
+
+    template <_RANGES input_range _Range, class _FormatContext>
+        requires _Is_specialization_v<typename _FormatContext::iterator, back_insert_iterator>
+              && derived_from<typename _FormatContext::iterator::container_type, _Fmt_buffer<_CharT>>
+    _FormatContext::iterator _Format(_Range&& _Rng, _FormatContext& _Ctx) const {
+        auto _Format_specs = _Specs;
+        if (_Specs._Dynamic_width_index >= 0) {
+            _Format_specs._Width =
+                _STD _Get_dynamic_specs<_Width_checker>(_Ctx.arg(static_cast<size_t>(_Specs._Dynamic_width_index)));
+        }
+
+        basic_string<_CharT> _Buffer;
+        {
+            _Fmt_iterator_buffer<back_insert_iterator<basic_string<_CharT>>, _CharT> _Fmt_buf(
+                _STD back_inserter(_Buffer));
+            using _Inserter      = back_insert_iterator<_Fmt_buffer<_CharT>>;
+            auto _Nested_context = basic_format_context<_Inserter, _CharT>::_Make_from(
+                _Inserter{_Fmt_buf}, _Ctx._Get_args(), _Ctx._Get_lazy_locale());
+
+            if constexpr (same_as<_Ty, _CharT>) {
+                if (_Specs._Type == 's' || _Specs._Type == '?') {
+                    _Format_as_string(_STD forward<_Range>(_Rng), _Nested_context, _Specs._Type == '?');
+                } else {
+                    _Format_as_sequence(_STD forward<_Range>(_Rng), _Nested_context);
+                }
+            } else {
+                _Format_as_sequence(_STD forward<_Range>(_Rng), _Nested_context);
+            }
+        }
+
+        const int _Width = static_cast<int>(_Buffer.size()); // TRANSITION, GH-4631
+        return _STD _Write_aligned(
+            _Ctx.out(), _Width, _Format_specs, _Fmt_align::_Left, [&](_FormatContext::iterator _Out) {
+                return _STD _Fmt_write(_STD move(_Out), basic_string_view{_Buffer});
+            });
+    }
+
+    template <_RANGES input_range _Range, class _FormatContext>
+    void _Format_as_sequence(_Range&& _Rng, _FormatContext& _Ctx) const {
+        _Ctx.advance_to(_STD _Fmt_write(_Ctx.out(), _Opening_bracket));
+        bool _Separate = false;
+        for (auto&& _Elem : _Rng) {
+            if (_Separate) {
+                _Ctx.advance_to(_STD _Fmt_write(_Ctx.out(), _Separator));
+            }
+
+            _Separate = true;
+            _Ctx.advance_to(_Underlying.format(_Elem, _Ctx));
+        }
+
+        (void) _STD _Fmt_write(_Ctx.out(), _Closing_bracket);
+    }
+
+    template <_RANGES input_range _Range, class _FormatContext>
+    void _Format_as_string(_Range&& _Rng, _FormatContext& _Ctx, bool _Debug) const {
+        if constexpr (_RANGES contiguous_range<_Range>) {
+            using _String_view = basic_string_view<_CharT>;
+            const auto _Size   = _STD _To_unsigned_like(_RANGES distance(_Rng));
+            if (_Size >= (numeric_limits<typename _String_view::size_type>::max)()) [[unlikely]] {
+                _Throw_format_error("Formatted range is too long.");
+            }

```

Comment:
Does this deserve `if constexpr`?
Also, I believe we can simply spell `size_t` for the size type ([[string.view.template.general]](https://eel.is/c++draft/string.view.template.general)).